core/consensus/aura/Cargo.toml

```diff
@@ -37,3 +37,11 @@ test-client = { package = "substrate-test-runtime-client", path = "../../test-ru
 tokio = "0.1.22"
 env_logger = "0.7.0"
 tempfile = "3.1.0"
+
+[features]
+default = ["full_crypto"]
+
+# This feature enables all crypto primitives for `no_std` builds like microcontrollers 
+# or Intel SGX. 
+# For the regular wasm runtime builds this should not be used.
+full_crypto = []
```

Comment:
That is actually not good.

You should change the macro to generate different output, if the `full_crypto` feature is added to the `primtives` crate.

(You probably need to create 2 different macros to make this possible)

Reply:
hmm, but wouldn't that change be more invasive and less understandable than pseudo-adding this feature to each crate using the macro?
Is your goal to leave the crates entirely untouched that have no use for the full_crypto feature?

Reply:
I don't want to add a feature, just because I want to use a macro. If the user does not provides the feature, the compiler will report weird errors. Macros should be self contained when possible and it is possible in this case.

Reply:
alright. this feels like manually doing the job cargo should do. But I agree.

Reply:
Yeah, I understand your pain, but I don't want to distribute this pain to everyone using this :D

Reply:
I think I've found a better way:
```
#[cfg(any(feature = "full_crypto", feature = "std"))]
```
because `std` always implies `full_crypto` we can safely put this logic. In the case of `babe` there is no `full_crypto`feature, but `std` will be there, given that you could accept putting this in all crates that use the app_crypto macro:

```
[features]
default = ["std"]
```

Reply:
just pushed my suggestion